crypto/ec/ecx_meth.c

```diff
@@ -855,6 +855,25 @@ static const EVP_PKEY_METHOD ed448_pkey_meth = {
 #ifdef S390X_EC_ASM
 # include "s390x_arch.h"
 
+/*
+ * constant-time conditional swap of the len-byte buffers a and b,
+ * depending on cond in {0,1}.
+ */
+static inline void s390x_cswap(unsigned char *a, unsigned char *b, size_t len,
+                               unsigned int cond)
+{
+    size_t i;
+    unsigned char tmp;
+    unsigned char mask = 0 - (unsigned char)cond;
+
+    for (i = 0; i < len; i++) {
+        tmp = a[i] ^ b[i];
+        tmp &= mask;
+        a[i] ^= tmp;
+        b[i] ^= tmp;
+    }
+}
```

Comment:
It seems like this function belongs in `constant_time.h`. We already have uint32_t and uint64_t versions. This just adds another variant.

Reply:
okay, i will move it there.

Reply:
done